File: interest.c
double monthlyInterest(float balance, float rate, int days)
{
    int i;
    double interest;
    double totalInterest = 0;
    double dailyRate = rate/365;

    for (i = 0; i < days; i++) {
        interest = balance * dailyRate;
        totalInterest += interest;
        balance += interest;
    }

    return totalInterest;
}

int main(int argc, char *argv[])
{
    int i;
    int niters = 4000000;

    for (i = 0; i < niters; i++)
        monthlyInterest(50000, 0.03, 30);

    return 0;
}

File: interest.py
# -*- coding: utf-8 -*-

def monthly_interest(starting_balance, rate, days=30):
	total_interest = 0
	daily_rate = rate/365
	balance = starting_balance
	
	for i in range(days):
		interest = balance * daily_rate
		total_interest += interest
		balance += interest
	
	return total_interest

if __name__ == '__main__':
    for i in range(100000):
        monthly_interest(50000, 0.03)
